### PR TITLE
feat: add liquidity provision SLA params to spot governance proposals…

### LINKED ISSUE #8731: [Feature]: SLA governance support
# Feature Overview

**In order to** create spot market with liquidity provision
**We will** add liquidity provision SLA params to governance proposals of new/update spot market
**So that** SLA liquidity engine can be initialised 

## Specs
- [[lp_mechanics](https://github.com/vegaprotocol/specs/blob/cosmicelevator/protocol/0044-LIME-lp_mechanics.md#market-parameters)]

### PR DESCRIPTION
**Pull request by ze97286**
… and market

Closes #8731

### CODE REVIEW

File: commands/proposal_submission.go [hunk 1]
```diff
@@ -1092,6 +1092,44 @@ func checkNewRiskParameters(config *protoTypes.NewMarketConfiguration) Errors {
...
+		errs.AddForProperty(fmt.Sprintf("%s.price_range", parent), ErrIsNotValidNumber)
+	} else if lppr.IsNegative() || lppr.IsZero() {
+		errs.AddForProperty(fmt.Sprintf("%s.price_range", parent), ErrMustBePositive)
+	} else if lppr.GreaterThan(num.DecimalFromInt64(100)) {
+		errs.AddForProperty(fmt.Sprintf("%s.price_range", parent), ErrMustBeAtMost100)
+	}
+
+	committmentMinTimeFraction, err := num.DecimalFromString(config.CommittmentMinTimeFraction)
+	if err != nil {
+		errs.AddForProperty(fmt.Sprintf("%s.committmentMinTimeFraction", parent), ErrIsNotValidNumber)
```
Comment (jeremyletang) [inline commands/proposal_submission.go:1112]: Not a big deal, but we usually name fields in snake_case in the errors.

File: commands/proposal_submission.go [hunk 2]
```diff
@@ -1092,6 +1092,44 @@ func checkNewRiskParameters(config *protoTypes.NewMarketConfiguration) Errors {
...
+	}
+
+	slaCompetitionFactor, err := num.DecimalFromString(config.SlaCompetitionFactor)
+	if err != nil {
+		errs.AddForProperty(fmt.Sprintf("%s.slaCompetitionFactor", parent), ErrIsNotValidNumber)
+	} else if slaCompetitionFactor.IsNegative() || slaCompetitionFactor.GreaterThan(num.DecimalOne()) {
+		errs.AddForProperty(fmt.Sprintf("%s.slaCompetitionFactor", parent), ErrMustBeWithinRange01)
+	}
+
+	// TODO are there constraints for performanceHysteresisEpochs?
```
Comment (jeremyletang) [inline commands/proposal_submission.go:1128]: To be don ein this PR or later? Does htat needs a spec updte?

File: core/execution/engine.go [hunk 3]
```diff
@@ -206,6 +206,14 @@ func (e *Engine) ReloadConf(cfg Config) {
...
 }
 
+func (e *Engine) OnEpochEvent(ctx context.Context, epoch types.Epoch) {
+	// TODO
+}
+
+func (e *Engine) OnEpochRestore(ctx context.Context, epoch types.Epoch) {
+	// TODO
+}
+
```
Comment (jeremyletang) [inline core/execution/engine.go:216]: Notthin to do the for not too? is that for a followup PR?

File: commands/proposal_submission.go (hunk 2 above)
Comment (ze97286) [reply commands/proposal_submission.go:1128]: as far as I can see there are no constraints for the market params in the spec other than the ones enforced here, so will require a spec update and possibly adding this later.

File: core/execution/engine.go (hunk 3 above)
Comment (ze97286) [reply core/execution/engine.go:216]: yes, in the spots PR this will call all markets but this is in the following spot pr

**Review by jeremyletang — COMMENTED** [review]
Is there any snapshots state to be handled as part of this PR seeing that the SpotMarket is getting some new fields (not sure the snapshots is being covered for spots market already anyway).

**Comment by ze97286** [comment]
> Is there any snapshots state to be handled as part of this PR seeing that the SpotMarket is getting some new fields (not sure the snapshots is being covered for spots market already anyway).

I don't think so. This should be encapsulated in types.Market which is saved in the snapshot in its proto counterpart and the conversion handles the new fields.

**Comment by jeremyletang** [comment]
> > Is there any snapshots state to be handled as part of this PR seeing that the SpotMarket is getting some new fields (not sure the snapshots is being covered for spots market already anyway).
> 
> I don't think so. This should be encapsulated in types.Market which is saved in the snapshot in its proto counterpart and the conversion handles the new fields.

Ok cool.

File: commands/proposal_submission.go (hunk 1 above)
Comment (ze97286) [reply commands/proposal_submission.go]: renamed

**Review by jeremyletang — DISMISSED** [review]

File: protos/sources/vega/governance.proto [hunk 4]
```diff
@@ -69,6 +69,8 @@ message NewSpotMarketConfiguration {
   }
   // Decimal places for order sizes, sets what size the smallest order / position on the spot market can be.
   int64 position_decimal_places = 6;
+  // Specifies the liquidity provision SLA paramters.
```
Comment (candida-d) [inline protos/sources/vega/governance.proto:72]:
```suggestion
  // Specifies the liquidity provision SLA parameters.
```

File: protos/sources/vega/governance.proto [hunk 5]
```diff
@@ -180,6 +182,8 @@ message UpdateSpotMarketConfiguration {
     // Log normal risk model parameters, valid only if MODEL_LOG_NORMAL is selected.
     LogNormalRiskModel log_normal = 101;
   }
+  // Specifies the liquidity provision SLA paramters.
```
Comment (candida-d) [inline protos/sources/vega/governance.proto:185]:
```suggestion
  // Specifies the liquidity provision SLA parameters.
```

File: protos/sources/vega/governance.proto [hunk 6]
```diff
@@ -456,6 +460,10 @@ enum ProposalError {
   PROPOSAL_ERROR_GOVERNANCE_CANCEL_TRANSFER_PROPOSAL_INVALID = 51;
   // Proposal for updating market state is invalid
   PROPOSAL_ERROR_INVALID_MARKET_STATE_UPDATE = 52;
+  // Liquidity provision SLA params are invalid
```
Comment (candida-d) [inline protos/sources/vega/governance.proto:463]:
```suggestion
  // Liquidity provision SLA parameters are invalid
```

File: protos/sources/vega/governance.proto [hunk 7]
```diff
@@ -456,6 +460,10 @@ enum ProposalError {
...
+  PROPOSAL_ERROR_INVALID_SLA_PARAMS = 53;
+  // Mandatory Liquidity provision SLA params are missing
```
Comment (candida-d) [inline protos/sources/vega/governance.proto:465]:
```suggestion
  // Mandatory liquidity provision SLA parameters are missing
```

File: protos/sources/vega/markets.proto [hunk 8]
```diff
@@ -305,6 +318,8 @@ message Market {
   optional string insurance_pool_fraction = 16;
   // ID of the market that succeeds this market if it exists. This will be populated by the system when the successor market is enabled.
   optional string successor_market_id = 17;
+  // LiquiditySLAParameters parameters for the market.
```
Comment (candida-d) [inline protos/sources/vega/markets.proto:321]:
```suggestion
  // Liquidity SLA parameters for the market.
```

**Review by jeremyletang — DISMISSED** [review]

**Review by jeremyletang — DISMISSED** [review]

**Review by candida-d — APPROVED** [review]

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**